src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-cluster-form/nfs-cluster-form.component.ts

```diff
@@ -0,0 +1,110 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
+import _ from 'lodash';
+import { UntypedFormControl } from '@angular/forms';
+import { CdFormGroup } from '~/app/shared/forms/cd-form-group';
+import { NfsService } from '~/app/shared/api/nfs.service';
+import { getFsalFromRoute, getPathfromFsal } from '../utils';
+import { SUPPORTED_FSAL } from '../models/nfs.fsal';
+import { NfsRateLimitComponent } from '../nfs-rate-limit/nfs-rate-limit.component';
+
+import { concat as observableConcat, Observable } from 'rxjs';
+import { NotificationType } from '~/app/shared/enum/notification-type.enum';
+import { NotificationService } from '~/app/shared/services/notification.service';
+import { NFSBwIopConfig } from '../models/nfs-cluster-config';
+import { CdForm } from '~/app/shared/forms/cd-form';
+
+@Component({
+  selector: 'cd-nfs-cluster-form',
+  templateUrl: './nfs-cluster-form.component.html',
+  styleUrls: ['./nfs-cluster-form.component.scss']
+})
+export class NfsClusterFormComponent extends CdForm implements OnInit {
+  @ViewChild(NfsRateLimitComponent, { static: false })
+  nfsRateLimitComponent!: NfsRateLimitComponent;
+
+  action: string;
+  nfsForm: CdFormGroup;
+  resource: string;
+  isEdit = false;
+  disablePerShare = true;
+  disablePerClient = true;
+  disablePerSharePerClient = true;
+  fsal: SUPPORTED_FSAL;
+  id: string;
+  qosValue: string;
+  nfsBwIopConfig: NFSBwIopConfig;
+  submitObservables: Observable<Object>[] = [];
+
+  constructor(
+    public actionLabels: ActionLabelsI18n,
+    private route: ActivatedRoute,
+    private router: Router,
+    private nfsService: NfsService,
+    private notificationService: NotificationService
+  ) {
+    super();
+    this.resource = $localize`Cluster`;
+  }
+  ngOnInit(): void {
```

Comment:
its a minor nit, but again its nice to leave an empty line before the function definition. I will write a lint rule to catch it every PR rather than me complaining on those.